pallets/pallet-deposit-storage/src/lib.rs

```diff
@@ -143,10 +143,19 @@ pub mod pallet {
 	/// Storage of all deposits. Its first key is a namespace, and the second
 	/// one the deposit key. Its value includes the details associated to a
 	/// deposit instance.
+	// SAFETY of Twox64Concat:
+	// The Namespace is not chosen by the user but by the pallet and therefore
+	// doesn't allow for arbitrary data.
 	#[pallet::storage]
 	#[pallet::getter(fn deposits)]
-	pub(crate) type Deposits<T> =
-		StorageDoubleMap<_, Twox64Concat, <T as Config>::Namespace, Twox64Concat, DepositKeyOf<T>, DepositEntryOf<T>>;
+	pub(crate) type Deposits<T> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		<T as Config>::Namespace,
+		Blake2_128Concat,
```

Comment:
note this change. I think the DepositKey might be chosable by the user

Reply:
This follows the same reasoning as my other comments (not sure if above or below). The concrete type of a deposit key is defined by the runtime, but it could take user-specified values, so I am all in favor for this change.